### PR TITLE
fix typo in duplicate did detection and update selecting data_dict if only 1 entry in the dict

### PR DESCRIPTION
**Pull request by NicholasArmour**
This fixes master not working

### CODE REVIEW

File: chipsec/chipset.py [hunk 1]
```diff
@@ -544,7 +544,7 @@ def init( self, platform_code, req_pch_code, start_driver, driver_exists=None, t
                 self.pch_rid = 0xFF
 
         if self.pch_vid == VID_INTEL and self.pch_did in pch_dictionary:
-            data_dict           = pch_dictionary[self.pch_did]
+            data_dict           = pch_dictionary[self.pch_did][0]
```
Comment (npmitche) [inline chipsec/chipset.py]: Are you guaranteed to only have one element in each PCH?

Comment (NicholasArmour) [reply chipsec/chipset.py]: Yes. The pch_dictionary definitions in chipset.py only have 1 list per pch_did key.

Comment (BrentHoltsclaw) [reply chipsec/chipset.py]: At this time the answer is yes only 1.  We will need to update the logic if this changes.

Comment (npmitche) [reply chipsec/chipset.py]: It might be good to check there's only one and if not, log a warning/info?

Comment (BrentHoltsclaw) [reply chipsec/chipset.py]: @npmitche could you create a pull request or issue to add the logging

**Pull request merged**